graylog2-web-interface/src/components/common/mode-pipeline.js

```diff
@@ -0,0 +1,103 @@
+import 'brace/mode/java';
+
+class PipelineHighlightRules extends window.ace.acequire('ace/mode/text_highlight_rules').TextHighlightRules {
+  constructor() {
+    super();
+
+    const keywords = 'let|rule|match|pipeline';
+
+    const operators = 'and|or|not';
+
+    const builtinConstants = 'all|either|during|when|then|end';
+
+    const builtinFunctions = 'to_bool|to_double|to_long|to_string|to_map|is_bool|is_number|is_double|is_long|is_string|'
```

Comment:
@kyleknighted What do you think about making this list dynamic? We could create an API endpoint that returns all builtin functions. That way we can't forget to add a function here.

And since pipeline functions are pluggable, using a hardcoded list would only work for the core functions. With a dynamic list, functions that are provided by plugins would also be properly colored.

Reply:
> We could create an API endpoint
Probably not necessary, we could use the same endpoint that populates the documentation on the pipeline rules page

> pipeline functions are pluggable
Great to know!

> making this list dynamic
I think it's a great idea, I'll toy with the possibilities of that... not sure how feasible it is considering Ace Editor needs that data when it loads.